### PR TITLE
feat(octavia): added l7 rule creation page

### PR DESCRIPTION
**Pull request by JacquesLarique**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/octavia-policies-and-rules`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-12735
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~ (n/a)

## Description

Added the L7 rule creation page

### CODE REVIEW

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/routing.js [hunk 1]
```diff
@@ -27,6 +27,24 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          coreURLBuilder.buildURL(
+            'public-cloud',
+            '#/pci/projects/:serviceName/octavia-load-balancer/:region/:loadbalancerId/listeners/:listenerId/l7/:policyId/rules/create',
+            {
+              serviceName: projectId,
+              region,
+              loadbalancerId,
+              listenerId,
+              policyId,
+            },
```
Comment (dectotam) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/routing.js:46]:
```suggestion
            '#/pci/projects/${projectId}/octavia-load-balancer/${region}/${loadbalancerId}/listeners/${listenerId}/l7/${policyId}/rules/create'
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/rules/components/rule-form/template.html [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            data-items="$ctrl.ruleTypes"
+            data-value-property="value"
+            data-match="label"
+            data-on-change="$ctrl.resetCompareType()"
+            data-placeholder="{{:: 'octavia_load_balancer_create_l7_rule_type_default' | translate }}"
+        >
+        </oui-select>
+        <span
+            class="help-block"
+            ng-if="$ctrl.model.ruleType"
```
Comment (dectotam) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/rules/components/rule-form/template.html:20]:
```suggestion
            data-ng-if="$ctrl.model.ruleType"
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/rules/components/rule-form/template.html [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            data-items="$ctrl.COMPARE_TYPES_AVAILABILITY_BY_TYPE[$ctrl.model.ruleType]"
+            data-value-property="value"
+            data-match="label"
+            data-on-change="$ctrl.refreshValue()"
+            data-placeholder="{{:: 'octavia_load_balancer_create_l7_rule_compare_type_default' | translate }}"
+        >
+        </oui-select>
+        <span
+            class="help-block"
+            ng-if="$ctrl.model.compareType"
```
Comment (dectotam) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/rules/components/rule-form/template.html:42]:
```suggestion
            data-ng-if="$ctrl.model.compareType"
```

File: packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/rules/components/rule-form/template.html [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        data-label-popover="{{:: 'octavia_load_balancer_create_l7_rule_key_tooltip' | translate }}"
+        data-error-messages="{ pattern: ('octavia_load_balancer_create_l7_rule_key_pattern' | translate)}"
+    >
+        <div class="oui-input-overlay">
+            <input
+                type="text"
+                class="oui-input"
+                name="key"
+                data-ng-required="$ctrl.RULE_TYPES_WITH_KEY.includes($ctrl.model.ruleType)"
+                data-pattern="^[a-zA-Z0-9!#$%&'*+-.^_`|~]+$"
```
Comment (dectotam) [inline packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/rules/components/rule-form/template.html:59]: why not add this regex to the constant file also ?

**Review by dectotam — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**